File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int denum1, int num1, int denum2, int num2) {
        int[] answer = new int [2];
        //분모
        int num = num1*num2;
        //분자
        int denum = denum1*num2+denum2*num1;
        
        int max=Math.max(num,denum);
        int min=Math.min(num,denum);
        
        //최대공약수(유클리드 호제법)
        //int a,int b 나머지=r 
        //while(b!=0){
        // int r = a%b;
        // a = b;
        // b = r;
        // }
        while(min!=0){
            int gcd = max % min;
            max = min;
            min = gcd;
        }
        
        answer[0] = denum/max;
        answer[1] = num/max;
        
        return answer;
    }
}

File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(denum1, num1, denum2, num2) {
  var answer = [];
  let topNum = (denum1 * num2) + (denum2 * num1);
  let downNum = num1 * num2;
  let maxNum = 1;
  for (let i = 1; i <= topNum; i++) {
    if (topNum % i === 0 && downNum % i === 0) {
      maxNum = i;
    }
  }
  return [topNum / maxNum, downNum / maxNum];
}
